mate-panel/menu.c

```diff
@@ -1238,13 +1229,9 @@ create_applications_menu (const char *menu_file,
 	g_signal_connect (menu, "button_press_event",
 			  G_CALLBACK (menu_dummy_button_press_event), NULL);
 
-	matemenu_tree_add_monitor (tree,
-			       (MateMenuTreeChangedFunc) handle_matemenu_tree_changed,
-			       menu);
-	g_signal_connect (menu, "destroy",
```

Comment:
Shouldn't we keep some cleanup function? Like disconnecting the signal handler when `menu` goes away.

Reply:
Ok, now added disconnect signal handle when menu destroy.